app/src/main/java/com/shagalalab/constitution/ui/search/SearchResultFragment.kt

```diff
@@ -29,10 +29,17 @@ class SearchResultFragment : Fragment(R.layout.fragment_search_result) {
     private lateinit var navController: NavController
     private val models: MutableList<ListItem> = mutableListOf()
     private var currentLanguageId = 0
+    private var resultsFor = ""
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        tvResults.text = getString(R.string.results_for, safeArgs.query)
+        when (safeArgs.lang) {
```

Comment:
You can move `resultsFor = ` out of when

Reply:
Actually, you don't need any variable, you can directly assign the result of `when`  to `tvResults.text`